src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Diagnostics/RazorDiagnosticsPublisher.cs

```diff
@@ -168,25 +158,72 @@ internal void ClearClosedDocuments()
         }
     }
 
+    private void ClearClosedDocumentsPublishedDiagnostics<T>(Dictionary<string, IReadOnlyList<T>> publishedDiagnostics) where T : class
```

Comment:
Consider making this a local function in `ClearClosedDocuments`, since it's only called from there and is assumed to be called under a lock.